CDCgov/prime-reportstream
# Experience/6523

This PR removes usage of NewApi/UseOrganizationResource/UseOrgName with a new UseOrganizationSettings hook that relies on useAuthorizedFetch (based on react-query).

Test Steps:
1. Log into reportstream
2. Set simple_report from organization list
3. Go to Submissions page
4. Verify organization description near page header
5. Go to Upload & Validate > Validate page
6. Verify organization description near page header
7. Go to Daily Data page
8. Verify organization description near page header

## Changes
- Created UseOrganizationSettings hook with test that relies on useAuthorizedFetch (react-query)
- Replaced usage of NewApi/UseOrganizationResource/UseOrgName with UseOrganizationSettings
- Marked older functions as deprecated

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #6523 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #6527
